src/router_node.c

```diff
@@ -1884,3 +1885,12 @@ void qd_link_restart_rx(qd_link_t *in_link)
         qd_connection_invoke_deferred(in_conn, deferred_AMQP_rx_handler, safe_ptr);
     }
 }
+
+
+// prepend formatted connection and link info to policy denial text and log it
+void qd_connection_log_policy_denial(qd_link_t *link, const char *text)
+{
+    qdr_link_t *rlink = (qdr_link_t*) qd_link_get_context(link);
```

Comment:
The qdr_link_t has a separate lifecycle from the qd_link_t.  IOW you'll want to check that rlink is not null here.